src/authn/index.js

```diff
@@ -601,8 +602,17 @@ function parseCognitoGroups(cognitoGroups) {
   const [internalGroups, nextstrainGroupsRoles] = partition(cognitoGroups, g => g.startsWith("!"));
 
   return {
-    // Just the Nextstrain Group names (i.e. excluding the "/role" suffix)
-    groups: [...new Set(nextstrainGroupsRoles.map(g => splitGroupRole(g).group))],
+    groups: [
+      // Just the Nextstrain Group names (i.e. excluding the "/role" suffix)
+      ...new Set(nextstrainGroupsRoles.map(g => splitGroupRole(g).group))
+    ].map(name => {
+      try {
+        return new Group(name);
+      } catch (error) {
+        // Nextstrain Group does not exist for the Cognito group prefix
+        return null;
+      }
+    }).filter(group => group !== null),
```

Comment:
This type change from `Array[String]` → `Array[Group]` will break the old-school session serialization/deserialization. I don't think we _need_ that older way of handling sessions any more, but it's currently still present.

Reply:
I'll look into ripping it out.

Reply:
https://github.com/nextstrain/nextstrain.org/pull/840 (but needs testing first)

Reply:
Rebased onto latest `master` to bring in changes from #840